### PR TITLE
New syndicate discounts (tm)

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

You good old friend Admiral Sabertooth has done it again, handing down all new and improved deals down to all field agents!
All you have to do is have some good old TC to spend and a premium syndicate uplink (tm) to gain these deals! (Not available to Nanotrasen employees).

Adds in the new deals:
 - Discount (60% chance): A simple discount, a percentage off the original price.
 - Buy 1 get 1 free (20% chance): Purchase the item and get a second one completely free! For when you want a second energy sword for no reason in particular!
 - Double Discount Bundle (20% chance): A discounted bundle containing 2 items, that's right 2! Get 2 random items with a reduced total price!

## Why It's Good For The Game

More deals, more fun.

## Changelog
:cl:
add: Adds in the buy 1 get 1 free syndicate uplink deal.
add: Adds in the double discount bundle syndicate uplink deal.
/:cl:

### CODE REVIEW

File: code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -63,20 +64,38 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
+				A.spawn_amount = 2
+			if(5)
+				//Get 2 items with their combined price reduced.
+				var/datum/uplink_item/second_I = pick_n_take(sale_items)
+				var/total_cost = second_I.cost + I.cost
+				var/discount = A.get_discount()
+				var/final_cost = max(round(total_cost * discount), 1)
+				//Setup the item
+				A.cost = final_cost
+				A.name += " + [second_I.name] (Discounted Bundle - [round(((initial(A.cost)-A.cost)/initial(A.cost))*100)]% off!)"
```
Comment (PowerfulBacon) [inline code/modules/uplink/uplink_items.dm]:
```suggestion
				A.name += " + [second_I.name] (Discounted Bundle - [round((total_cost / final_cost )*100)]% off!)"
```

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -118,9 +139,17 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	return pick(4;0.75,2;0.5,1;0.25)
 
 /datum/uplink_item/proc/purchase(mob/user, datum/component/uplink/U)
-	var/atom/A = spawn_item(item, user, U)
-	if(purchase_log_vis && U.purchase_log)
-		U.purchase_log.LogPurchase(A, src, cost)
+	//Spawn base items
+	for(var/i in 1 to spawn_amount)
```
Comment (ike709) [inline code/modules/uplink/uplink_items.dm:143]:
```suggestion
	for(var/i in 1 to spawn_amount)
```
Don't you need to specify the len? Or does BYOND do that implicitly?

Comment (PowerfulBacon) [reply code/modules/uplink/uplink_items.dm:143]: spawn_amount is a number

File: code/modules/uplink/uplink_items.dm [hunk 3]
```diff
@@ -118,9 +139,17 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
+		var/atom/A = spawn_item(item, user, U)
+		if(purchase_log_vis && U.purchase_log)
+			U.purchase_log.LogPurchase(A, src, cost)
+	//Spawn bonust items
+	if(islist(bonus_items))
+		for(var/bonus in bonus_items)
```
Comment (PowerfulBacon) [inline code/modules/uplink/uplink_items.dm:149]: AUTOREVIEW 9000: Check returned: Line 78: Formatting warning: Commas should have a space after them. (3, 5, 6) not (3,5,6 or 3 ,4 ,5)

File: code/modules/uplink/uplink_items.dm [hunk 4]
```diff
@@ -63,20 +64,38 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
-		A.cost = max(round(A.cost * discount),1)
-		A.name += " ([round(((initial(A.cost)-A.cost)/initial(A.cost))*100)]% off!)"
-		A.desc += " Normally costs [initial(A.cost)] TC. All sales final. [pick(disclaimer)]"
+		switch(rand(1, 5))
+			if(1 to 3)
+				//X% off!
+				var/discount = A.get_discount()
+				if(A.cost >= 20) //Tough love for nuke ops
+					discount *= 0.5
+				A.cost = max(round(A.cost * discount),1)
```
Comment (PowerfulBacon) [inline code/modules/uplink/uplink_items.dm]:
```suggestion
				A.cost = max(round(A.cost * discount), 1)
```

**Comment by heepox** [comment]
Ah yes TraitorMart has the best deals anywhere in the syndicate.

This is nice.

**Comment by Bastian0930** [comment]
I really love the idea! I've always thought there should be more types of discounts. Though, I don't see buy one get one being too useful most of the time. I still like the idea, though!

**Comment by FortSumter** [comment]
Since when did the syndicate become Walmart? Also you can't combine two eswords into a dual esword. That hasn't been a feature in a long time, unless I'm unaware of someone re-adding it.

**Comment by PowerfulBacon** [comment]
> Since when did the syndicate become Walmart? Also you can't combine two eswords into a dual esword. That hasn't been a feature in a long time, unless I'm unaware of someone re-adding it.

Since the syndicate realised that deals drive increased sales. They have started a new initiative to deliver great deals to the consumers! (All deals come from the returns section of the syndicate market, quality not guaranteed. No refunds)

**Review by utilbird — APPROVED** [review]

**Pull request merged**